src/scripting/lua_menu_item.cpp

```diff
@@ -24,26 +24,84 @@ static lg::log_domain log_scripting_lua("scripting/lua");
 #define LOG_LUA LOG_STREAM(info, log_scripting_lua)
 #define ERR_LUA LOG_STREAM(err, log_scripting_lua)
 
+static const char menuitemKey[] = "menu_item";
+
+void luaW_pushmenuitem(lua_State* L, gui2::menu_item& m)
+{
+	new(L) lua_ptr<gui2::menu_item>(m);
+	luaL_setmetatable(L, menuitemKey);
+}
+
+gui2::menu_item& luaW_checkmenuitem(lua_State* L, int n)
+{
+	lua_ptr<gui2::menu_item>& lp =  *static_cast<lua_ptr<gui2::menu_item>*>(luaL_checkudata(L, n, menuitemKey));
+	auto ptr = lp.get_ptr();
+	if(!ptr) {
+		luaL_argerror(L, n, "menu_item was deleted");
+	}
+	return *ptr;
+}
+
+lua_ptr<gui2::menu_item>& luaW_checkmenuitem_ptr(lua_State* L, int n)
+{
+	lua_ptr<gui2::menu_item>& lp =  *static_cast<lua_ptr<gui2::menu_item>*>(luaL_checkudata(L, n, menuitemKey));
+	auto ptr = lp.get_ptr();
+	if(!ptr) {
+		luaL_argerror(L, n, "menu_item was deleted");
+	}
+	return lp;
+}
+
 static int impl_menuitem_get(lua_State *L)
 {
-    config **cfg = static_cast<config **>(lua_touserdata(L, 1));
+    gui2::menu_item& item = luaW_checkmenuitem(L, 1);
+
     char const *m = luaL_checkstring(L, 2);
+/*
+	return_bool_attrib("checkbox", *(*ptr).checkbox);  // optional
+	//return_tstring_attrib("details", *((**cfg).details));  // optional
+	return_tstring_attrib("details", *(*ptr).details);  // optional
+	//return_string_attrib("icon", (**cfg).icon);
+	return_string_attrib("icon", (*ptr).icon);
+	//return_string_attrib("image", *(**cfg).image);  // optional
+	return_string_attrib("image", *(*ptr).image);  // optional
+	//return_tstring_attrib("label", (**cfg).label);
+*/
```

Comment:
Please don't leave commented code in. It's fine for now if the PR isn't quite finished and you need it for your reference, though.